Xamarin.Forms.Maps/Map.cs

```diff
@@ -155,37 +156,10 @@ void OnItemTemplatePropertyChanged(DataTemplate oldItemTemplate, DataTemplate ne
 
 		void OnItemsSourceCollectionChanged(object sender, NotifyCollectionChangedEventArgs e)
 		{
-			switch (e.Action)
-			{
-				case NotifyCollectionChangedAction.Add:
-					if (e.NewStartingIndex == -1)
-						goto case NotifyCollectionChangedAction.Reset;
-					foreach (object item in e.NewItems)
-						CreatePin(item);
-					break;
-				case NotifyCollectionChangedAction.Move:
-					if (e.OldStartingIndex == -1 || e.NewStartingIndex == -1)
-						goto case NotifyCollectionChangedAction.Reset;
-					// Not tracking order
-					break;
-				case NotifyCollectionChangedAction.Remove:
-					if (e.OldStartingIndex == -1)
-						goto case NotifyCollectionChangedAction.Reset;
-					foreach (object item in e.OldItems)
-						RemovePin(item);
-					break;
-				case NotifyCollectionChangedAction.Replace:
-					if (e.OldStartingIndex == -1)
-						goto case NotifyCollectionChangedAction.Reset;
-					foreach (object item in e.OldItems)
```

Comment:
With the extensions, a Replace with different item counts won't work the same here. That is, if you are replacing 3 items with 2 items, the old version would work; the new version will just jump to Reset. Is that something we should be concerned about?

Reply:
Well, I see #5582 got merged, so I guess if it was OK for BindableLayout, it should be OK for Map too? It makes sense to have INotifyCollectionChanged implementation consistent across controls. 
Maybe the extensions could be changed (a future PR?) so that it doesn't reset collection when new and old items differ in length.

Reply:
> I guess if it was OK for BindableLayout, it should be OK for Map too?

Either that, or we made a terrible mistake merging 5582. That implementation of Replace in `Apply` is very suspect.

It looks like Map.Pins is an `ObservableCollection<Pin>`, so while I don't like this implementation of `Apply`, it should be fine for this purpose. It looks like the implementation of `ObservableCollection` only uses `NotifyCollectionChangedAction.Replace` for a single item being set. So at least for this specific application, it shouldn't cause any immediate problems.

Reply:
Yes, most apps use `ObservableCollection` so it's fine, but for apps which which do something more advanced and throw custom `NotifyCollectionChangedAction` it will be less efficient